applications/nrf_desktop/src/modules/hid_forward.c

```diff
@@ -318,16 +404,177 @@ static u8_t hidc_read_cfg(struct bt_gatt_hids_c *hidc,
+		if (subscriber->timestamp == 0){
+			LOG_INF("DFU next forward");
+			err = change_connection_latency(subscriber, LATENCY_LOW);
+			if (err) {
+				LOG_WRN("Error while change of connection parameters");
+				return err;
+			}
+		}
+		subscriber->timestamp = k_uptime_get_32();
+
+		if (!k_delayed_work_remaining_get(&config_fwd_timeout)) {
+			k_delayed_work_submit(&config_fwd_timeout,
+					      FORWARD_WORK_DELAY);
+		}
+	}
+
+	return err;
+}
+
+static void handle_config_forward(const struct config_forward_event *event)
+{
+	struct hids_subscriber *subscriber = find_subscriber_hidc(event->recipient);
+
+	if (!subscriber) {
+		LOG_INF("Recipent %02" PRIx16 "not found", event->recipient);
+		return;
+	}
+
+	struct bt_gatt_hids_c *recipient_hidc =	&subscriber->hidc;
+
+	__ASSERT_NO_MSG(recipient_hidc != NULL);
+
+	if (!bt_gatt_hids_c_ready_check(recipient_hidc)) {
+		LOG_WRN("Cannot forward, peer disconnected");
+
+		notify_config_forwarded(CONFIG_STATUS_DISCONNECTED_ERROR);
+		return;
+	}
+
+	struct bt_gatt_hids_c_rep_info *config_rep =
+		bt_gatt_hids_c_rep_find(recipient_hidc,
+					BT_GATT_HIDS_REPORT_TYPE_FEATURE,
+					REPORT_ID_USER_CONFIG);
+
+	if (!config_rep) {
+		LOG_ERR("Feature report not found");
+		notify_config_forwarded(CONFIG_STATUS_WRITE_ERROR);
+		return;
+	}
+
+	if (event->dyndata.size > UCHAR_MAX) {
+		LOG_WRN("Event data too big");
+		return;
+	}
+
+	struct config_channel_frame frame;
+	u8_t report[REPORT_SIZE_USER_CONFIG];
+
+	if (event->status == CONFIG_STATUS_FETCH) {
+		LOG_INF("Forwarding fetch request");
+		frame.status = CONFIG_STATUS_FETCH;
+	}
+
+	frame.recipient = event->recipient;
+	frame.event_id = event->id;
+	frame.event_data_len = event->dyndata.size;
+	frame.event_data = (u8_t *)event->dyndata.data;
+
+	int err = switch_to_low_latency(subscriber);
+
+	if (err) {
+		notify_config_forwarded(CONFIG_STATUS_WRITE_ERROR);
+		return;
+	}
+
+	int pos = config_channel_report_fill(report, sizeof(report), &frame,
+					     false);
+
+	if (pos < 0) {
+		LOG_WRN("Could not set report");
+		return;
+	}
+
+	err = bt_gatt_hids_c_rep_write(recipient_hidc,
+				       config_rep,
+				       hidc_write_cb,
+				       report,
+				       sizeof(report));
+	if (err) {
+		LOG_ERR("Writing report failed, err:%d", err);
+		notify_config_forwarded(CONFIG_STATUS_WRITE_ERROR);
+	}
+
+	return;
+}
+
+static void handle_config_forward_get(const struct config_forward_get_event *event)
+{
+	struct hids_subscriber *subscriber =
+		find_subscriber_hidc(event->recipient);
+
+	if (!subscriber) {
+		LOG_INF("Recipent %02" PRIx16 "not found", event->recipient);
+		return;
+	}
+
+	struct bt_gatt_hids_c *recipient_hidc =	&subscriber->hidc;
+
+	__ASSERT_NO_MSG(recipient_hidc != NULL);
+
+	int err = switch_to_low_latency(subscriber);
+
+	if (err) {
+		notify_config_forwarded(CONFIG_STATUS_WRITE_ERROR);
+		return;
+	}
+
+	if (forward_pending) {
+		LOG_DBG("GATT read already pending");
+		return;
+	} else {
```

Comment:
no need for else (and increased indent) if we return in line 541

Reply:
done